common/scala/src/main/scala/whisk/core/entity/Exec.scala

```diff
@@ -93,7 +93,10 @@ protected[core] case class BlackBoxExec(image: String) extends Exec(Exec.BLACKBO
     override val sentinelledLogs = false
 }
 
-protected[core] case class SequenceExec(code: String, components: Vector[String]) extends Exec(Exec.SEQUENCE) {
+/**
+ * add temporary field that holds the "fixed" names for components where the '_' is replaced by the user's namespace
+ */
+protected[core] case class SequenceExec(code: String, components: Vector[String], fixedComponents: Option[Vector[String]] = None) extends Exec(Exec.SEQUENCE) {
```

Comment:
Eventually I'll store only the resolved ones. I don't want to touch Perry's code to make sure it still works as designed. On a second thought, you mentioned that it may not make a difference in this case. I'll try to change it to see if Perry's code still works.